azkaban-common/src/main/java/azkaban/project/ProjectManager.java

```diff
@@ -319,27 +320,67 @@ public void updateProjectDescription(final Project project, final String descrip
+    if (flowFile != null && flowFile.exists()) {
+      flowFile.delete();
+    }
+    return props;
   }
 
-  public Props getJobOverrideProperty(final Project project, final String jobName)
-      throws ProjectManagerException {
-    return this.projectLoader.fetchProjectProperty(project, jobName + ".jor");
+  public Props getProperties(final Project project, final Flow flow, final String jobName,
+      final String source) throws ProjectManagerException {
+    if (Double.compare(flow.getAzkabanFlowVersion(), Constants.AZKABAN_FLOW_VERSION_2_0) == 0) {
+      // Return the properties from the original uploaded flow file.
+      return getPropertiesFromFlowFile(flow, jobName, source, 1);
+    } else {
+      return this.projectLoader.fetchProjectProperty(project, source);
+    }
   }
 
-  public void setJobOverrideProperty(final Project project, final Props prop, final String jobName,
-      final User modifier)
-      throws ProjectManagerException {
-    prop.setSource(jobName + ".jor");
-    final Props oldProps =
-        this.projectLoader.fetchProjectProperty(project, prop.getSource());
+  public Props getJobOverrideProperty(final Project project, final Flow flow, final String jobName,
+      final String source) throws ProjectManagerException {
+    if (Double.compare(flow.getAzkabanFlowVersion(), Constants.AZKABAN_FLOW_VERSION_2_0) == 0) {
+      final int flowVersion = this.projectLoader
+          .getLatestFlowVersion(flow.getProjectId(), flow.getVersion(), source);
+      return getPropertiesFromFlowFile(flow, jobName, source, flowVersion);
+    } else {
+      return this.projectLoader.fetchProjectProperty(project, jobName + ".jor");
+    }
+  }
 
-    if (oldProps == null) {
-      this.projectLoader.uploadProjectProperty(project, prop);
+  public void setJobOverrideProperty(final Project project, final Flow flow, final Props prop,
+      final String jobName, final String source, final User modifier)
+      throws ProjectManagerException {
+    final Props oldProps;
+    if (Double.compare(flow.getAzkabanFlowVersion(), Constants.AZKABAN_FLOW_VERSION_2_0) == 0) {
+      final int flowVersion = this.projectLoader.getLatestFlowVersion(flow.getProjectId(), flow
+          .getVersion(), source);
+      final File flowFile = this.projectLoader.getUploadedFlowFile(flow.getProjectId(), flow
+          .getVersion(), flowVersion, source);
+      final String path = flow.getId() + Constants.PATH_DELIMITER + jobName;
+      oldProps = FlowLoaderUtils.getPropsFromYamlFile(path, flowFile);
+
+      FlowLoaderUtils.setPropsInYamlFile(path, flowFile, prop);
+      this.projectLoader
+          .uploadFlowFile(flow.getProjectId(), flow.getVersion(), flowVersion + 1, flowFile);
+      if (flowFile != null && flowFile.exists()) {
+        flowFile.delete();
+      }
     } else {
-      this.projectLoader.updateProjectProperty(project, prop);
+      prop.setSource(jobName + ".jor");
```

Comment:
what is ".jor" ? can we make it as a constant variable a meaningful name?

Reply:
I think it means job override. I can make it a constant variable.